src/main/java/org/kuali/kfs/module/cam/document/service/impl/GlLineServiceImpl.java

```diff
@@ -273,24 +273,44 @@ public CapitalAssetInformation findCapitalAssetInformation(String documentNumber
      * @param entry
      * @param capitalAssetLineType
      */
-    protected void addToCapitalAssetsMatchingLineType(List<CapitalAssetInformation> matchingAssets, CapitalAssetInformation capitalAsset, GeneralLedgerEntry entry) {
+    protected void addToCapitalAssetsMatchingLineType(List<CapitalAssetInformation> matchingAssets,
+            CapitalAssetInformation capitalAsset, GeneralLedgerEntry entry) {
         List<CapitalAssetAccountsGroupDetails> groupAccountLines = capitalAsset.getCapitalAssetAccountsGroupDetails();
 
         for (CapitalAssetAccountsGroupDetails groupAccountLine : groupAccountLines) {
-            if (groupAccountLine.getDocumentNumber().equals(entry.getDocumentNumber()) &&
-                    groupAccountLine.getChartOfAccountsCode().equals(entry.getChartOfAccountsCode()) &&
-                    groupAccountLine.getAccountNumber().equals(entry.getAccountNumber()) &&
-                    groupAccountLine.getFinancialObjectCode().equals(entry.getFinancialObjectCode())) {
-            	// check that debit matches target acct lines and credit matches source accounting lines
-            	if((StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.SOURCE_ACCT_LINE_TYPE_CODE) && StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_CREDIT_CODE))
-            			|| (StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.TARGET_ACCT_LINE_TYPE_CODE) && StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_DEBIT_CODE))){
-            	matchingAssets.add(capitalAsset);
-            	  break;
-              }
+            if (groupAccountLine.getDocumentNumber().equals(entry.getDocumentNumber())
+                    && groupAccountLine.getChartOfAccountsCode().equals(entry.getChartOfAccountsCode())
+                    && groupAccountLine.getAccountNumber().equals(entry.getAccountNumber())
+                    && groupAccountLine.getFinancialObjectCode().equals(entry.getFinancialObjectCode())) {
+                // check that debit matches target acct lines and credit matches
+                // source accounting lines
+                boolean isGroupLineSource = StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.SOURCE_ACCT_LINE_TYPE_CODE);
+                boolean isGroupLineTarget = StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.TARGET_ACCT_LINE_TYPE_CODE);
+                boolean isGLEntryCredit = StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_CREDIT_CODE);
+                boolean isGLEntryDebit = StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_DEBIT_CODE);
+                boolean isErrorCorrection = groupAccountLine.getAmount().isNegative();
```

Comment:
Just to confirm, is this a sufficient check for seeing if the accounting line is an error correction? I'm not familiar enough with these documents to know whether non-error-correction lines are always positive (even if they're credits), so I just wanted to make sure.